spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/DefaultHealthIndicatorRegistry.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.health;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.util.Assert;
+
+/**
+ * Default implementation of {@link HealthIndicatorRegistry}.
+ *
+ * @author Vedran Pavic
+ */
+public class DefaultHealthIndicatorRegistry implements HealthIndicatorRegistry {
+
+	private final Map<String, HealthIndicator> healthIndicators =
```

Comment:
Since operations on ```healthIndicators``` map require external synchronization anyway (see ```register``` operation as the most obvious example), using a ```ConcurrentHashMap``` would introduce an unnecessary overhead.

See comments on outdated diffs for more background.